src/display/editor/tools.js

```diff
@@ -59,6 +60,112 @@ class IdManager {
   }
 }
 
+/**
+ * Class to manage the images used by the editors.
+ * The main idea is to try to minimize the memory used by the images.
+ * The images are cached and reused when possible
+ * We use a refCounter to know when an image is not used anymore but we need to
+ * be able to restore an image after a remove+undo, so we keep a file reference
+ * or an url one.
+ */
+class ImageManager {
+  #baseId = getUuid();
+
+  #id = 0;
+
+  #cache = null;
+
+  async #get(key, rawData) {
+    this.#cache ||= new Map();
+    let data = this.#cache.get(key);
+    if (data === null) {
+      return data;
+    }
+    if (data?.bitmap) {
```

Comment:
Looking at similar code in the `getFromId`-method, could we perhaps simplify this a tiny bit?
```suggestion
    if (!data) {
      return null;
    }
    if (data.bitmap) {
```

Reply:
When data is null it means that we already tryed to load the image but for any reasons it failed.